server/src/types.rs

```diff
@@ -50,6 +50,14 @@ impl EdgeToken {
             alias: None,
         }
     }
+
+    pub fn subsumes(&self, other: &EdgeToken) -> bool {
+        return self.token_type == other.token_type
+            && self.environment == other.environment
+            && (self.projects.contains(&"*".into())
+                || (self.projects.len() >= other.projects.len()
+                    && other.projects.iter().all(|p| self.projects.contains(p))));
```

Comment:
I hate doing 2 iterations on projects, especially because the length check was supposed to prevent going through the list in some situations